java/integTest/src/test/java/glide/SharedClientTests.java

```diff
@@ -177,4 +185,85 @@ public void inflight_requests_limit(boolean clusterMode, int inflightRequestsLim
+        } else {
+            assertEquals(
+                    "OK", ((GlideClusterClient) client).configSet(Map.of("requirepass", password)).get());
+        }
+    }
+
+    @SneakyThrows
+    private void dropConnection(BaseClient client, boolean disconnect) {
+        var command =
+                disconnect ? new String[] {"CLIENT", "KILL", "TYPE", "NORMAL"} : new String[] {"RESET"};
+        if (client instanceof GlideClient) {
+            ((GlideClient) client).customCommand(command).get();
+        } else {
+            ((GlideClusterClient) client).customCommand(command, ALL_NODES).get();
+        }
+    }
+
+    private static Stream<Arguments> clientsAndReconnectMode() {
+        return Stream.of(
+                Arguments.of(standaloneClient, true),
+                Arguments.of(standaloneClient, false),
+                Arguments.of(clusterClient, true),
+                Arguments.of(clusterClient, false));
+    }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("clientsAndReconnectMode")
+    public void password_update(BaseClient client, boolean reconnect) {
+        var key = UUID.randomUUID().toString();
+        var pwd = UUID.randomUUID().toString();
+        client.set(key, "meow meow").get();
+
+        try (BaseClient testClient =
+                client instanceof GlideClient
+                        ? GlideClient.createClient(commonClientConfig().build()).get()
+                        : GlideClusterClient.createClient(commonClusterClientConfig().build()).get()) {
+
+            // validate that we can get the value
+            assertEquals("meow meow", testClient.get(key).get());
+
+            // set the password and forcefully drop connection for the second client
+            setPassword(client, pwd);
+            dropConnection(client, reconnect);
+
+            // client should reconnect, but will receive NOAUTH error
+            var exception = assertThrows(ExecutionException.class, () -> testClient.get(key).get());
+            assertInstanceOf(RequestException.class, exception.getCause());
+
+            assertEquals("OK", testClient.updateConnectionPassword(pwd, RE_AUTHENTICATE).get());
+
+            // after setting new password we should be able to work with the server
+            assertEquals("meow meow", testClient.get(key).get());
+
+            // unset the password and drop connection again
+            setPassword(client, pwd);
+            dropConnection(client, reconnect);
+
+            // client should reconnect, but will receive NOAUTH error
+            exception = assertThrows(ExecutionException.class, () -> testClient.get(key).get());
+            assertInstanceOf(RequestException.class, exception.getCause());
+
+            assertEquals(
+                    "OK",
+                    testClient
+                            .updateConnectionPassword(reconnect ? USE_ON_NEW_CONNECTION : RE_AUTHENTICATE)
```

Comment:
Problematic terminology, you make it seems like its one of the two. Use on new connection is always true.